zebra-chain/src/transaction/serialize.rs

```diff
@@ -400,13 +397,14 @@ impl ZcashSerialize for Transaction {
         //
         // Since we checkpoint on Sapling activation, we won't ever need
         // to check the smaller pre-Sapling transaction size limit.
+        writer.write_u32::<LittleEndian>(self.header())?;
+
```

Comment:
I think it would be preferable not to have common code between the transaction serialization and the sighash implementation, because I think that both implementations are clearer and easier to understand when you can immediately see all the data being written and in what order.